lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -334,6 +334,19 @@ class ReductionOpConversionPattern : public OpConversionPattern<TTIROpTy> {
   LogicalResult
   matchAndRewrite(TTIROpTy op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
+    int64_t inputTensorRank =
```

Comment:
Please move this check into op verifier. We don't want to allow a creation of such an op standalone. Conversion will fail as well, because this check will fail during the coverted op creation.